Documentation/RepoApi.md

```diff
@@ -0,0 +1,137 @@
+
+This list of actions can be added to in the future as more scenarios are implemented.
+
+source-build formats arguments using a style that is compatible with MSBuild. The reasoning for this is that the
+majority of repos have an MSBuild based build system. Using this format will allow for arguments to easily be passed
+through to MSBuild. Similarly, both action and parameter arguments are case-insensitive, just like MSBuild default
+behavior.
+
+Actions will take the form of `-t:ActionName`, just like MSBuild targets are passed to `msbuild.exe`.
+
+### Parameters
+
+Each action above will take a set of parameters used to pass information into the action and control its behavior.
+
+Parameters will take the form of `-p:ParameterName=ParameterValue`, just like MSBuild properties are passed to `msbuild.exe`.
+One slight variation here is the usage of `-p` instead of `/p`.  MSBuild supports both forms, but source-build chooses `-p` because
+it meets [POSIX recommendations](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html) for command line parameters
+and avoids any confusion with file paths on Unix machines.
+
+In order to avoid conflicts between existing MSBuild properties used by the repos and the properties passed in by source-build,
+source-build will append a prefix to every property it passes in: `DotNet`.
+
+#### Detecting build from source
+
+The first parameter used by source-build will be a value indicating whether the build is being performed in "source only" mode. This
+means repos cannot use tools/packages/references that haven't already been built by the current build.
+
+`-p:DotNetBuildFromSource=true`
+
+Some repos build more projects than is necessary to build the .NET Core stack. For example, the `dotnet/roslyn` repo builds
+the C# compiler, which is needed, but it also builds Visual Studio extensions, which aren't needed. These VS extensions
+cannot be built without using an external dependency on tooling that isn't available when building from source.
+
+#### Available tools
+
+In order for repos to build, they need tools in order to do the build (ex. MSBuild). These tools are not generally available
+on all supported operating systems. As such, many repos today have the ability to pull these tools from the internet during the
+build.
+
+However, there are a couple problems with each repo downloading its own toolset.
+
+1. Different versions can cause inconsistencies
+1. It takes a non-trivial amount of time to download the toolset, if every repo does it during the complete .NET Core build,
+it will add a lot of overhead to the build.
+
+To solve this, source-build will have a set of common tools available that it will pass into each repos build.
+
+First, a [.NET Core SDK](https://www.microsoft.com/net/download/core#/sdk) will be available for repos to be able to invoke
+MSBuild, NuGet, etc. The path to the folder containing the `dotnet` executable will be available using the
+
+`-p:DotNetToolDir`
```

Comment:
I generally prefer Path over Dir in msbuild properties but maybe that is just me.  Also think we should call this CLI so I would suggest DotNetCLIPath for this property.

Reply:
I've seen `Dir` in a lot of places in the repos, and IMO it's useful. I'd assume `DotNetCLIPath` is the path to the `dotnet` file, not the directory containing it.

However, I have seen `...File` used for files before, so it looks like we have competing standards. `Dir`/`Path`, `Path`/`File`, and (maybe) `Path` for both files and dirs.

Reply:
I much prefer Dir. There is zero ambiguity as to what it means

Reply:
You convinced me. Dir it is.

Reply:
These parameters are getting renamed to 

```
DotNetCoreSdkDir
DotNetBuildToolsDir
```